oedipus/__init__.py

```
@@ -219,6 +251,28 @@ class S(object):
 
     __len__ = count
 
+    def excerpt(self, result):
+        """Takes a result and returns the excerpt as a string."""
+        if self._results_cache is None:
+            # This guarantees that we have results before we try
+            # getting excerpts for results.
+            self._raw()
```

If it's not too much work, let's throw a DudeThisSHasNoResultsYet error. Hmm, now you've got my second-guessing our decision to not put excerpt() on the search result somehow....